test/e2etest/kubetest/xml_junit_result.go

```diff
@@ -0,0 +1,61 @@
+package kubetest
+
+import (
+	"encoding/xml"
+	"github.com/gardener/test-infra/test/e2etest/config"
+	"regexp"
+	"strings"
+)
+
+func (result *JunitXMLResult) CalculateAdditionalFields() {
+	result.SuccessfulTests = result.ExecutedTests - result.FailedTests
+	result.DurationInt = int(result.DurationFloat)
+	regexpSigGroup := regexp.MustCompile(`^\[.*?]`)
+	for i, _ := range result.Testcases {
+		result.Testcases[i].calculateAdditionalFields(regexpSigGroup)
+		// TODO: why is here result.Testcases[i].calculateAdditionalFields(regexpSigGroup) not possible, why is assigning to variable is required?
+	}
+}
+
+func (testcase *TestcaseResult) calculateAdditionalFields(regexpSigGroup *regexp.Regexp) {
+	testcase.SigGroup = regexpSigGroup.FindString(testcase.Name)
+	if testcase.SkippedRaw != nil {
+		testcase.Skipped = true
+	}
+	if testcase.FailureText == "" {
+		testcase.Status = Success
+		testcase.Successful = true
+	} else {
+		testcase.Status = Failure
+		testcase.Successful = false
+	}
+	testcase.DurationInt = int(testcase.DurationFloat)
+	testcase.TestDesc = config.DescriptionFile
+	testcase.ExecutionGroup = strings.Join(config.TestcaseGroup, ",")
+}
+
+type JunitXMLResult struct {
```

Comment:
comment which fields are calculated and which should be initialized.

Maybe create a `func New() *JunitXMLResult` function so that you can be sure that everything you need is defined.